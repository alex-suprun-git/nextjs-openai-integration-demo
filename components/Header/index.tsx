'use client';

import { usePathname } from 'next/navigation';
import Link from 'next/link';
import { FaQuestionCircle } from 'react-icons/fa';
import { useTranslations } from 'next-intl';

const Header = ({
  userPromptLimit,
  userPromptUsed,
  userPromptLimitRenewal,
}: {
  userPromptLimit: string;
  userPromptUsed: string;
  userPromptLimitRenewal: string;
}) => {
  const path = usePathname();
  const isActiveLink = (href: string) => path === href;

  const t = useTranslations('Header');

  const navigationLinks = [
    { label: t('navigation.journal'), href: '/journal' },
    { label: t('navigation.statistics'), href: '/statistics' },
  ];

  return (
    <>
      <div className="mb-12 flex md:mb-0">
        {navigationLinks.map((link) => (
          <Link
            className={isActiveLink(link.href) ? 'font-bold' : ''}
            key={link.href}
            href={link.href}
          >
            <span className="mr-12 block">{link.label}</span>
          </Link>
        ))}
      </div>
      <div className="ml-auto flex flex-col items-center text-center md:mr-10 md:flex-row md:text-left">
        <p className="mr-2 leading-6">
          {t.rich('labels.promptSymbolsRemaining', {
            userPromptUsed,
            userPromptLimit,
            strong: (chunks) => <strong>{chunks}</strong>,
          })}
        </p>
        <div
          className="tooltip tooltip-bottom mt-4 md:tooltip-left md:mt-0"
          data-tip={t('labels.promptSymbolsRenewalDate', { userPromptLimitRenewal })}
        >
          <span className="cursor-pointer">
            <FaQuestionCircle />
          </span>
        </div>
      </div>
    </>
  );
};

export default Header;
