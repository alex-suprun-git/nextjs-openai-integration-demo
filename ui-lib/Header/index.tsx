const Header = ({ children }: { children: React.ReactNode }) => (
  <div
    data-testid="header"
    className="navbar border-black/10 bg-gray-900 px-10 py-5 md:items-center lg:h-[80px] lg:py-0"
  >
    {children}
  </div>
);
export default Header;
