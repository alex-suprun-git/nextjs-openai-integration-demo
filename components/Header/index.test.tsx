import { describe, it, expect, vi, Mock } from 'vitest';
import { render, screen } from '@testing-library/react';
import { usePathname } from 'next/navigation';
import Header from './';

vi.mock('next/navigation', () => ({
  usePathname: vi.fn(),
}));

describe('Header', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('renders navigation links correctly', () => {
    (usePathname as Mock).mockReturnValue('/journal');
    const { container } = render(<Header userPromptLimit="1000" userPromptUsed="500" />);
    expect(container).toMatchSnapshot();

    expect(screen.getByText('Journal')).toBeInTheDocument();
    expect(screen.getByText('Statistics')).toBeInTheDocument();
  });

  it('applies active class to the current path link', () => {
    (usePathname as Mock).mockReturnValue('/journal');
    render(<Header userPromptLimit="1000" userPromptUsed="500" />);

    expect(screen.getByRole('link', { name: 'Journal' })).toHaveClass('font-bold');
    expect(screen.getByRole('link', { name: 'Statistics' })).not.toHaveClass('font-bold');
  });

  it('displays the correct prompt usage', () => {
    (usePathname as Mock).mockReturnValue('/journal');
    render(<Header userPromptLimit="1000" userPromptUsed="500" />);

    expect(screen.getByText('500/1000')).toBeInTheDocument();
    expect(screen.getByText('prompt symbols remaining', { exact: false })).toBeInTheDocument();
  });
});
