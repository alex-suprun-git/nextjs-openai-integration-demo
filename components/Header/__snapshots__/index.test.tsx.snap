// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders navigation links correctly 1`] = `
<div>
  <div
    class="mb-12 flex md:mb-0"
  >
    <a
      class="font-bold"
      href="/journal"
    >
      <span
        class="mr-12 block"
      >
        Journal
      </span>
    </a>
    <a
      class=""
      href="/statistics"
    >
      <span
        class="mr-12 block"
      >
        Statistics
      </span>
    </a>
  </div>
  <div
    class="ml-auto mr-10 flex"
  >
    <p>
      You have
       
      <strong>
        500
        /
        1000
      </strong>
       
      prompt symbols remaining
    </p>
  </div>
</div>
`;
